projects/rero/ng-core/src/lib/record/search/aggregation/buckets/buckets.component.ts

```diff
@@ -0,0 +1,219 @@
+
+@Component({
+  selector: 'ng-core-record-search-aggregation-buckets',
+  templateUrl: './buckets.component.html'
+})
+export class BucketsComponent implements OnInit, OnDestroy {
+  /**
+   * Buckets list for aggregation
+   */
+  @Input()
+  buckets: Array<any>;
+
+  /**
+   * Aggregation key
+   */
+  @Input()
+  aggregationKey: string;
+
+  /**
+   * Bucket size, if not null, reduce displayed items to this size.
+   */
+  @Input()
+  size: number;
+
+  /**
+   * More and less on aggregation content (facet)
+   */
+  moreMode = true;
+
+  /**
+   * Current selected values for the aggregation
+   */
+  aggregationsFilters: Array<AggregationsFilter> = [];
+
+  /**
+   * Subscription to aggregationsFilters observable
+   */
+  private _aggregationsFiltersSubscription: Subscription;
+
+  constructor(
+    private translate: TranslateService,
+    private _recordSearchService: RecordSearchService
+  ) { }
+
+  ngOnInit() {
+    this._aggregationsFiltersSubscription = this._recordSearchService.aggregationsFilters.subscribe(
+      (aggregationsFilters: Array<AggregationsFilter>) => {
+        this.aggregationsFilters = aggregationsFilters;
+      }
+    );
+  }
+
+  ngOnDestroy() {
+    this._aggregationsFiltersSubscription.unsubscribe();
+  }
+
+  /**
+   * Returns selected filters for the aggregation key.
+   * @return List of selected filters
+   */
+  get aggregationFilters(): Array<string> {
+    const aggregationFilters = this.aggregationsFilters.find((item: AggregationsFilter) => item.key === this.aggregationKey);
+    if (aggregationFilters === undefined) {
+      return [];
+    }
+    return aggregationFilters.values;
+  }
+
+  /**
+   * Get current language
+   * @return Current language
+   */
+  get language(): string {
+    return this.translate.currentLang;
+  }
+
+  /**
+   * Return bucket size
+   * @return Bucket size
+   */
+  get bucketSize(): number {
+    const size = this.buckets.length;
+
+    if (this.size === null || this.moreMode === false) {
+      return size;
+    }
+
+    return this.size;
+  }
+
+  /**
+   * Check if a value is present in selected filters
+   * @param value - string, filter value
+   * @return Boolean
+   */
+  isSelected(value: string): boolean {
+    return this.aggregationFilters.includes(value);
+  }
+
+  /**
+   * Update selected filters by adding or removing the given value and push
+   * values to service.
+   * @param value - string, filter value
+   */
+  updateFilter(bucket: any) {
+    const index = this.aggregationsFilters.findIndex((item: any) => {
+      return item.key === this.aggregationKey;
+    });
+
+    // Aggregation is not yet selected
+    if (index === -1) {
+      this._recordSearchService.updateAggregationFilter(this.aggregationKey, [bucket.key]);
+    } else {
+      // Bucket is not yet selected, we add value to selected values
+      let values = this.aggregationsFilters[index].values;
+
+      if (!values.includes(bucket.key)) {
+        values.push(bucket.key);
```

Comment:
what's happen if in the url only the child filter is present?

Reply:
The records are well filtered by the child filter, but the parent's levels are not selected. How can we handle this case, in your opinion?

Reply:
I left a TODO in the record search service to remember to handle this case.